Princeton-CDH/geniza
# #235 - Add display_label field to document type, update index and views

### What this PR does
- Per #235:
  - Adds optional `display_label` field to `DocumentType` model
  - Updates `DocumentType` `__str__` to use `display_label` if available
  - Adds data migration to add the provided labels
  - Updates all instances of "Unknown" to "Unknown type"
  - Updates `Document` solr index to use the new `__str__` method
  - Fixes tests with duplicate DocumentTypes
  - Adds SASS mixin for properly formatted doctype in titles, used in both search and detail view

### Steps to test

1. Visit the document search page
2. Verify that document titles display in the format "**Document type**; shelfmark"
3. Verify that the document types display according to the labels specified in #235 
4. Verify that documents with unknown type read "Unknown type", rather than just "Unknown"
4. Navigate to a document detail page
5. Verify that 2-4 remain true on this page

### Deploy notes 

- Solr reindexing is required on deploy.
